tests/src/test/scala/com/lightbend/paradox/markdown/IncludeDirectiveSpec.scala

```diff
@@ -50,4 +50,29 @@ class IncludeDirectiveSpec extends MarkdownBaseSpec {
       |<p>This file should be included by IncludeDirectiveSpec</p>""")
   }
 
+  it should "include nested code snippets" in {
+    markdown("""@@include(tests/src/test/resources/include-code-snip.md)""") shouldEqual html("""
+      |<pre class="prettyprint">
+      |<code class="language-conf">
+      |a = b</code>
+      |</pre>""")
+  }
+
+  it should "include headers from nested snippets in the toc" in {
+    markdown(
+      """
+        |@@toc { depth=2 }
+        |
+        |# Hello
+        |
+        |@@include(tests/src/test/resources/headers.md)
+        |""") should include(html("""
+           |<div class="toc">
+           |  <ul>
+           |    <li><a href="test.html#heading-1" class="header">Heading 1</a></li>
+           |    <li><a href="test.html#heading-2" class="header">Heading 2</a></li>
+           |  </ul>
+           |</div>"""))
```

Comment:
Is it OK, that `# Hello` is not included in the toc?

Reply:
No, it's not, because Paradox strips the first header that it finds in the markdown from the page (regardless of where it is in the page) and treats that as the page title and heading. The behaviour is a bit weird when you don't put it right at the top, but all pages should start with a header so it's not a problem in practice.

That said, your comment caused me to look deeper into the handling, and I realised that toc directives (not navigation indexes) are quite broken because when they calculate if a header should be included in the page or not, they look at whether its index in the file is below the index of the toc directive. Of course, if the header is in an included file, or the toc is in an included file and the header is in the outer file, what needs to be compared is the include index to the header or toc index, since their locations in different files are not comparable. So, I've pushed a commit that fixes this - it includes a path of include indexes with every header, and adds the include index path to the rendering context, and the calculation has been updated to use this path in its considerations.